components/brave_rewards/resources/page/components/adsBox.tsx

```diff
@@ -91,9 +91,38 @@ class AdsBox extends React.Component<Props, State> {
   getAdsSubdivisions = () => {
     const {
       adsSubdivisionTargeting,
-      automaticallyDetectedAdsSubdivisionTargeting
+      automaticallyDetectedAdsSubdivisionTargeting,
     } = this.props.rewardsData.adsData
 
+    const {
+      currentCountryCode
+    } = this.props.rewardsData
+
+    let subdivisions: Array<[string, string]> = []
+    if (currentCountryCode === 'US') {
+      subdivisions = this.getUnitedStatesSubdivisions()
+    } else if (currentCountryCode === 'CA') {
+      subdivisions = this.getCanadaSubdivisions()
+    }
+
+    if (adsSubdivisionTargeting === 'DISABLED') {
+      subdivisions.unshift(['DISABLED', getLocale('adsSubdivisionTargetingDisabled')])
+    } else {
+      subdivisions.unshift(['DISABLED', getLocale('adsSubdivisionTargetingDisable')])
+    }
+
+    const subdivisionMap = new Map<string, string>(subdivisions)
+    const subdivision = subdivisionMap.get(automaticallyDetectedAdsSubdivisionTargeting)
+    if (subdivision && adsSubdivisionTargeting === 'AUTO') {
+      subdivisions.unshift(['AUTO', getLocale('adsSubdivisionTargetingAutoDetectedAs', { adsSubdivisionTarget: subdivision })])
+    } else {
+      subdivisions.unshift(['AUTO', getLocale('adsSubdivisionTargetingAutoDetect')])
+    }
+
+    return subdivisions
+  }
+
+  getUnitedStatesSubdivisions = () => {
```

Comment:
I see that following the existing conventions, we're defining these subdivisions in at least 3 different places. Would it be possible or desirable to  return these subdivisions (along with names) from the ads service, and then have the front-end use that?

Reply:
@aseren @zenparsing would be great for the ads lib to return these values so that we can add future regions without the need to change ts files etc. how about we merge this change as it will help the sales team and as the next task we make this change. Your thoughts? Thanks

Reply:
Sounds good to me!

Reply:
@zenparsing @aseren raised https://github.com/brave/brave-browser/issues/25249